pkg/utils/config.go

```diff
@@ -0,0 +1,133 @@
+package utils
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	color "github.com/threeal/bro/pkg/utils/color"
+)
+
+type Config interface {
+	Read() error
+	Write() error
+	init()
+}
+
+var HOME_DIR = getHomeDirectory()
+var BRO_CONFIG_DIR = filepath.Join(*HOME_DIR, ".bro")
+
+type BackendConfig struct {
+	ListenAddr *string `json:"listen_addr"`
+}
+
+type ClientConfig struct {
+	BackendAddr *string `json:"backend_addr"`
+}
+
+func getHomeDirectory() *string {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		log.Fatalf("failed to determine home directory: %v", err)
+	}
+	return &homeDir
+}
+
+func (c *BackendConfig) Read() error {
+	return readConfigFromFile(c, "backend_config.json")
+}
+
+func (c *ClientConfig) Read() error {
+	return readConfigFromFile(c, "config.json")
+}
+
+func (c *BackendConfig) Write() error {
+	return writeConfigToFile(c, "backend_config.json")
+}
+
+func (c *ClientConfig) Write() error {
+	return writeConfigToFile(c, "config.json")
+}
+
+func (c *BackendConfig) init() {
+	if c.ListenAddr == nil {
+		reader := bufio.NewReader(os.Stdin)
+		fmt.Print(color.Gray + "question" + color.Reset + " listen address (:320): ")
+		text, _ := reader.ReadString('\n')
+		if text == "\n" {
+			text = ":320"
+		}
+		c.ListenAddr = &text
+	}
+}
+
+func (c *ClientConfig) init() {
+	if c.BackendAddr == nil {
+		reader := bufio.NewReader(os.Stdin)
+		fmt.Print(color.Gray + "question" + color.Reset + " backend address (localhost:320): ")
+		text, _ := reader.ReadString('\n')
+		if text == "\n" {
+			text = "localhost:320"
+		}
+		c.BackendAddr = &text
+	}
+}
+
+func NewBackendConfig() *BackendConfig {
+	return &BackendConfig{ListenAddr: nil}
+}
+
+func NewClientConfig() *ClientConfig {
+	return &ClientConfig{BackendAddr: nil}
+}
+
+func InitializeBackendConfig() *BackendConfig {
+	backendConfig := NewBackendConfig()
+	initializeConfig(backendConfig)
+	return backendConfig
+}
+
+func InitializeClientConfig() *ClientConfig {
+	clientConfig := NewClientConfig()
+	initializeConfig(clientConfig)
+	return clientConfig
+}
+
+func initializeConfig(c Config) {
+	if err := c.Read(); err != nil {
+		if err = c.Write(); err != nil {
+			log.Fatalf("failed to initialize config: %v", err)
+		}
+	}
+	c.init()
+	c.Write()
+}
+
+func writeConfigToFile(c Config, configName string) error {
+	file, _ := json.MarshalIndent(c, "", "  ")
+	configPath := filepath.Join(BRO_CONFIG_DIR, configName)
```

Comment:
so.. since `BRO_CONFIG_DIR` may return error because `HOME_DIR` may return error because `getHomeDirectory` may return error, we may need to modify the logic here. E.g.

```go
func getHomeDir() (string, error) {
  ...
}

func getConfigDir() (string, error) {
  homeDir, err := getHomeDir()
  if err != nil {
    return "", err
  }
  ...
}

func WriteConfigToFile(c Config, configName string) error {
  ...
  configDir, err := getConfigDir()
  if err != nil {
    return err
  }
  ...
}
```